### ISSUE TITLE
[Improvement]: Validity Checks

### ISSUE THREAD

**Issue by dpfaffenbauer** [pimcore/pimcore#16460]
### Improvement description

Pimcore always loads all Data in order to check for validity. If a certain field has not been loaded (isLazyLoaded) it is not necessary to check for validity right? (https://github.com/pimcore/pimcore/blob/11.x/models/DataObject/Concrete.php#L129)

We got the issue that on every save our custom extensions are forced to load data, even though it is not necessary since nothing changed or nothing has been set. I think there is some room for improvement.

Maybe we could also introduce a Validation Interface to disable that completely. The CoreShop Extensions don't really need that, they are validated on save anyawy.

**Comment by fashxp** [comment]
probably related: https://github.com/pimcore/pimcore/issues/4154

**Comment by dpfaffenbauer** [comment]
@fashxp yep

**Comment by fashxp** [comment]
if I remember correctly, we also need to load the data to be able to write the versions ... but I'm not sure though anymore...

**Comment by dpfaffenbauer** [comment]
writing the version can be offloaded to a messenger task, so not a big deal

**Comment by fashxp** [comment]
not so sure about that when thinking about concurrency and consistency aspects

**Comment by dpfaffenbauer** [comment]
right. Still, when versioning is disabled it is not necessary to load everything
